File: NodeJS/ReverseNumber/ReverseNumber.js
/**
 * @description   Reverse Number 
 * @param         val {int}
 * @returns       int
 */
const reverseNumber = (val) => {
  let reverse = 0;
  while (val > 0){
    remainder = val % 10;
    reverse   = reverse * 10 + remainder;
    val       = Math.floor( val / 10 );
  }
  return reverse;
}

// Test Value //
console.log(reverseNumber(1234));     // 4321   (√)
console.log(reverseNumber(100002));   // 200001 (√)
console.log(reverseNumber(2000));     // return 2 (next improvement)

File: Python/ReverseNumber/ReverseNumber.py
"""
==============
Reverse Number
==============
example:
Input : 1234
Output: 4321

note: input is integer and output must be integer (NOT string)
"""
class Solution:
  def reverseNumber(val):
    reverse = 0
    while (val > 0):
      remainder   = val % 10
      reverse     = reverse * 10 + remainder
      val         = val // 10
    return reverse

##############
# Test Value #
##############
if __name__ == "__main__":
  print(Solution.reverseNumber(1234))
  print(Solution.reverseNumber(56374))
  print(Solution.reverseNumber(12034))
